packages/snowpack/src/build/build-import-proxy.ts

```diff
@@ -65,7 +67,20 @@ export function wrapHtmlResponse({
   code = code.replace(/\/?%PUBLIC_URL%\/?/g, config.buildOptions.baseUrl);
 
   if (hmr) {
-    code += `<script type="module" src="${getMetaUrlPath('hmr.js', isDev, config)}"></script>`;
+    const hmrScript = `<script type="module" src="${getMetaUrlPath(
+      'hmr.js',
+      isDev,
+      config,
+    )}"></script>`;
+
+    const closingBodyMatch = code.match(CLOSING_BODY_TAG);
+    if (closingBodyMatch && closingBodyMatch[0]) {
+      // if </body> tag, append before that ends
+      code = code.replace(new RegExp(`(${closingBodyMatch[0]})`), `${hmrScript}$1`);
+    } else {
+      // if no </body> tag (technically not required), append to end
+      code += hmrScript;
```

Comment:
Fix #2: add code within `<body>`

Reply:
yea, tbh this feels more error-prone than leaving as-is. If we ever run into a browser not loading it properly I'd be happy to revisit, but until then I think it's safe to assume that this is okay (since it's dev only).

Reply:
I hear you but HMR is a core feature, and something we want to be stable. It feels weird shipping knowingly-incorrect HTML to the browser 100% of the time (current behavior). Is there a better way we can implement this?

Reply:
100% agreed in principle, but in practice this seems to introduce new edge cases without fixing any known issue. 

If you could solve the comment issue by only injecting into the body if you find one and only one closing `</body>` tag, then I'd feel a bit better.

Reply:
Makes sense. You’re right—this isn’t solving an issue, it just bugs me quite a bit that our default behavior relies on a browser quirk. Updated the code to silently fallback to the previous behavior if multiple `</body>` tags are detected (either in comments or without).

I don’t want to introduce an HTML parser just for this—that feels like overkill at this point (though if we needed it for something else, I’d definitely use that here).

Reply:
awesome, yea that seems like a great tradeoff to me.